scripts/build_osm_network.py

```diff
@@ -395,11 +425,89 @@ def get_transformers(buses, lines):
     return df_transformers
 
 
+def get_converters(buses, links):
+    """
+    Function to create fake converter links that connect buses of the same station_id of different polarities
+    """
+
+    df_converters = []
+
+    for g_name, g_value in buses.sort_values("voltage", ascending=True).groupby(
+        by=["station_id"]
+    ):
+
+        # note: by construction there cannot be more that two buses with the same station_id and same voltage
+        n_voltages = len(g_value)
+
+        # Assuming that a converter stations should have both AC and DC parts
+        if g_value["dc"].any() & ~g_value["dc"].all():
+
+            for id in range(0, n_voltages - 1):
```

Comment:
With this procedure, you add a covnerter among multiple buses in the same substation, but we should add only a single converter for every dc bus between the two nodes with the closest voltage

Reply:
Should be fixed now along with a similar correction for transformers. A direct csvs check has been again very helpful :)